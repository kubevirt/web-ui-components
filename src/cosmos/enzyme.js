import createContext from 'react-cosmos-test/enzyme';

import paths from '../../config/paths';

export default function createTestContext(args) {
  args.proxies = args.proxies || []; // workaround of react-cosmos-test bug
  return createContext({
    ...args,
    cosmosConfigPath: `${paths.config}/cosmos.config.js`,
  });
}
