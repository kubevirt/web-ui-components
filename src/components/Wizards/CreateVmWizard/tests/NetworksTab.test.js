import React from 'react';
import { shallow } from 'enzyme';
import { NetworksTab } from '..';
import NetworksTabFixture from '../fixtures/NetworksTab.fixture';

const testNetworksTab = () => <NetworksTab {...NetworksTabFixture.props} />;

describe('<NetworksTab />', () => {
  it('renders correctly', () => {
    const component = shallow(testNetworksTab());
    expect(component).toMatchSnapshot();
  });
});
