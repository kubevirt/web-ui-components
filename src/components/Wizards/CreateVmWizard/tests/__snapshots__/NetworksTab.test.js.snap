// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NetworksTab /> renders correctly 1`] = `
<TableFactory
  actionButtons={
    Array [
      Object {
        "className": "create-network",
        "disabled": false,
        "id": "create-network-btn",
        "onClick": [Function],
        "text": "Create NIC",
      },
    ]
  }
  columns={
    Array [
      Object {
        "hasAddendum": true,
        "header": Object {
          "label": "NIC Name",
          "props": Object {
            "style": Object {
              "width": "32%",
            },
          },
        },
        "property": "name",
        "renderConfigs": Array [
          Object {
            "id": "name-edit",
            "type": "text",
          },
        ],
      },
      Object {
        "header": Object {
          "label": "MAC Address",
          "props": Object {
            "style": Object {
              "width": "32%",
            },
          },
        },
        "property": "mac",
        "renderConfigs": Array [
          Object {
            "id": "mac-edit",
            "type": "text",
          },
        ],
      },
      Object {
        "header": Object {
          "label": "Network Configuration",
          "props": Object {
            "style": Object {
              "width": "32%",
            },
          },
        },
        "property": "network",
        "renderConfigs": Array [
          Object {
            "choices": Array [
              "pxe-net-conf",
              "Pod Networking",
            ],
            "id": "network-edit",
            "initialValue": "--- Select Network Definition ---",
            "type": "dropdown",
          },
        ],
      },
      Object {
        "header": Object {
          "props": Object {
            "style": Object {
              "width": "4%",
            },
          },
        },
        "renderConfigs": Array [
          Object {
            "actions": Array [
              Object {
                "actionType": "delete",
                "text": "Remove NIC",
              },
            ],
            "id": "actions",
            "type": "actions",
            "visibleOnEdit": false,
          },
        ],
      },
    ]
  }
  error="At least one bootable NIC has to be defined for PXE boot"
  onRowActivate={[Function]}
  onRowDeleteOrMove={[Function]}
  onRowUpdate={[Function]}
  rows={Array []}
/>
`;
